### DISCUSSION TITLE
use values from extraEnvs in helmrelease

### DISCUSSION THREAD

**Discussion by jimmi4u** [fluxcd/flux2#3658] [Q&A]
Hi everybody,

I'm trying to get pgbouncer running and am struggling with that section under values in my helmrelease.yaml: 

extraEnvs:
      - name: admin_users
        valueFrom:
          secretKeyRef:
            key: user
            name: lime-pguser-lime
      - name: adminPassword
         valueFrom:
           secretKeyRef:
             key: password
             name: lime-pguser-lime
     - name: dbname
        valueFrom: 
          secretKeyRef:
            key: dbname
            name: lime-pguser-lime
    - name: host
       valueFrom: 
          secretKeyRef:
            key: host
            name: lime-pguser-lime

so those values have to be mapped to some variables in the config section, like that:
config: 
  adminPassword: {{  .Values.extraEnvs.adminPassword | b64dec }}  
( tried also with {{ tpl .Values.extraEnvs.adminPassword.valueFrom.secretKeyRef.key | b64dec }}   )

which in that case is not working: 
yaml: invalid map key: map[string]interface {}{".Values.extraEnvs.adminPassword | b64dec":""} and 
Kustomization reconciliation failed: kustomize build failed: map[string]interface {}(nil): yaml: invalid map key: map[string]interface {}{"tpl .Values.extraEnvs.adminPassword | b64dec":"" }

Is there a proper/correct way for doing that. ( chatGPT didn't help ;) ) and I didn't find any further information....
Many thanks for your help!

**Comment by kingdonb** [comment]
It is very difficult to read what you've done with this formatting.

If you can edit the response to use:
```
\```
```
backticks around the code sections (three backticks above and below, in markdown formatting), it will be much easier to read.

It is generally not needed to base64 decode secrets by hand. If you have a secret base64 encoded, that is the natural state of the secret. The values will be base64 decoded when they arrive. That might be all you needed to hear.

> so those values have to be mapped to some variables in the config section, like that:

It isn't clear what this means at all. If the chart accepts these values, then I'd expect it to know what to do with them without further additional configuration being required. Maybe it was just the base64 issue that blocked everything.

Please let us know if this helps!
